app/routes/index.tsx

```diff
@@ -1,151 +1,43 @@
-// import { getCollections } from "~/models/collection.client";
-// import Main from "~/components/main/Main";
-// import Layout from "~/components/Layout";
-// import { Outlet } from "@remix-run/react";
+import { perks } from "mocks/DUMMY_DATA";
 
-// export default function Index() {
-//   useEffect(() => {});
-//   return (
-//     <Layout>
-//     </Layout>
-//   );
-// }
-
-import { perks, collections, trendingProducts } from "mocks/DUMMY_DATA";
+import {
+  getAllCollections,
+  getTrendingProducts,
+} from "~/models/collection.client";
+import { useState, useEffect } from "react";
 
 export default function Home() {
-  return (
-    <main>
-      <Collections />
-      <Trending />
-    </main>
-  );
-}
+  const [loading, setLoading] = useState(true);
+  const [collections, setCollections] = useState([]);
+  const [trendingProducts, setTrendingProducts] = useState([]);
 
-function Trending() {
-  return (
-    <>
-      <section aria-labelledby="trending-heading">
-        <div className="mx-auto max-w-7xl py-24 px-4 sm:px-6 sm:py-32 lg:px-8 lg:pt-32">
-          <div className="md:flex md:items-center md:justify-between">
-            <h2
-              id="favorites-heading"
-              className="text-2xl font-bold tracking-tight text-gray-900"
-            >
-              Trending Products
-            </h2>
-            <a
-              href="/"
-              className="hidden text-sm font-medium text-indigo-600 hover:text-indigo-500 md:block"
-            >
-              Shop the collection
-              <span aria-hidden="true"> &rarr;</span>
-            </a>
-          </div>
+  //Get Collections Data
+  useEffect(() => {
+    getAllCollections().then((data) => {
+      setCollections(data);
+      setLoading(false);
+    });
+  }, []);
 
-          <div className="mt-6 grid grid-cols-2 gap-x-4 gap-y-10 sm:gap-x-6 md:grid-cols-4 md:gap-y-0 lg:gap-x-8">
-            {trendingProducts.map((product) => (
-              <div key={product.id} className="group relative">
-                <div className="h-56 w-full overflow-hidden rounded-md group-hover:opacity-75 lg:h-72 xl:h-80">
-                  <img
-                    src={product.imageSrc}
-                    alt={product.imageAlt}
-                    className="h-full w-full object-cover object-center"
-                  />
-                </div>
-                <h3 className="mt-4 text-sm text-gray-700">
-                  <a href={product.href}>
-                    <span className="absolute inset-0" />
-                    {product.name}
-                  </a>
-                </h3>
-                <p className="mt-1 text-sm text-gray-500">{product.color}</p>
-                <p className="mt-1 text-sm font-medium text-gray-900">
-                  {product.price}
-                </p>
-              </div>
-            ))}
-          </div>
+  // Get Trending Data
+  useEffect(() => {
+    getTrendingProducts().then((trends) => {
```

Comment:
why not in the same useEffect? combine them into one since the dependency array is the same.